### PR TITLE
Polls frontend

### PR DESCRIPTION
**Pull request by Piwakk**
Use the `Table` component for the polls tables.

### CODE REVIEW

File: frontend/src/services/api/polls.ts [hunk 1]
```diff
@@ -1,40 +1,104 @@
...
-        ),
+const listGeneric = (parameters: string = "") => (page = 1) =>
+    unwrap<PaginatedResponse<Poll[]>>(
+        apiService
+            .get<PaginatedResponse<Poll[]>>(
+                `/polls/?page=${page}${
+                    parameters === undefined ? "" : "&"
+                }${parameters}`
+            )
+            .then(parseDates)
```
Comment (abocquet) [inline frontend/src/services/api/polls.ts:35]: Does this return a promise?

Comment (Piwakk) [reply frontend/src/services/api/polls.ts:35]: It returns an `AxiosResponse`, like `apiService.get`

File: frontend/src/components/polls/polls_table/PollsTable.tsx [hunk 2]
```diff
@@ -1,91 +1,103 @@
...
+
+    const [stateFilter, setStateFilter] = useState<PollStateFilter>(
+        defaultStateFilter
     );
 
-    function Content() {
-        if (status === "loading") return <PollsLoading />;
-        else if (status === "success" && polls && user) {
-            return (
+    if (authService.isStaff || !adminVersion) {
```
[the remark this answers is not in the document]
Comment (Piwakk) [reply frontend/src/components/polls/polls_table/PollsTable.tsx:48]: Nice!

File: frontend/src/components/utils/table/TablePagination.tsx [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+                        className="page-link paginate_button previous"
+                        href="#"
+                        aria-label="Précédent"
+                    >
+                        <span aria-hidden="true">&laquo;</span>
+                    </a>
+                </li>
+                <li className="page-item">
+                    <a className="page-link paginate_button current" href="#">
+                        1
```
[the remark this answers is not in the document]
Comment (Piwakk) [reply frontend/src/components/utils/table/TablePagination.tsx:18]: Yes, I don't know why it's here. >:(

File: frontend/src/components/utils/table/Table.tsx [hunk 4]
```diff
@@ -9,9 +9,17 @@ import { Sorting } from "./sorting";
  * @param columns a list of `Column` objects. If the `Table` does not need a sorting, pass a list of `{ key, header, render }`
  * object; otherwise, it's more convenient to use `useColumns`.<br />
  * `key` is a string, unique among the other keys.<br />
+<<<<<<< HEAD
```
[the remark this answers is not in the document]
Comment (Piwakk) [reply frontend/src/components/utils/table/Table.tsx:12]: Indeed

File: frontend/src/services/api/polls.ts (hunk 1 above)
Comment (abocquet) [reply frontend/src/services/api/polls.ts:35]: `apiService.*.get` returns a `Promise<AxiosResponse<_>>`

**Pull request merged**